CE-10488: added changelog for CE 2022.1 (#41)
Esri/cityengine-sdk: changelog.md

# CITYENGINE SDK 2.7.85XX CHANGELOG

This section lists changes compared to CityEngine SDK 2.6.8300.

## General Info
* CityEngine SDK 2.7.85XX is used in CityEngine 2022.1.85XX.

## PRT API
* `prt::generate()`: fixed a (rare) crash on Linux.

## PRTX API

> * `prtx::EncodePreparator`: added `PrepartionFlags::meshMerging()` function and `MeshMerging::Action` enum. ALL_OF_SAME_MATERIAL_AND_TYPE corresponds to the old `mergeByMaterial()` flag; ALL_OF_SAME_TYPE is new functionality and allows for merging meshes with various materials into one mesh.  

Reviewer: ```suggestion
* `prtx::EncodePreparator`: added `PrepartionFlags::meshMerging()` function and `MeshMerging::Action` enum. `ALL_OF_SAME_MATERIAL_AND_TYPE` corresponds to the old `mergeByMaterial()` flag; `ALL_OF_SAME_TYPE` is new functionality and allows for merging meshes with various materials into one mesh.  
```

* `prtx::EncodePreparator`: deprecated `mergeByMaterial()` 

## CGA
* New operations:
  * splitAndSetbackPerimeter operation.
* Bugfixes:
  * offset operation: Fixed wrong geometry in case the inside selector was used but no inside faces were created due to a too large (negative) offset. Now the original geometry is kept in such cases.
  * deleteHoles operation: Fixed wrong geometry when deleting holes in non-planar shapes.
  * comp operation: Fixed wrong geometry when selecting face components with non-planar holes.
  * split operation: Fixed warnings containing the wrong operation name.
  * Local Edits:
    * Fixed a rare crash in deep recursive functions.
    * split, splitArea operation: Fixed a crash in nested repeat split patterns (repeat in repeat). 

## Built-In Codecs Changes and Fixes
* Added Decoder for Esri's lerc image format (com.esri.prt.codecs.LercDecoder).
* USD Codecs:

>   * Dynamic link libraires (e.g. usd_ms.dll and tbd.dll) are now prefixed with 'prt_' in order to prevent clashes if host app uses usd as well.

Reviewer: ```suggestion
  * The 3rd-party libraries (usd_ms.dll and tbb.dll) are now prefixed with 'prt_' in order to prevent clashes if these libraries are already present in a host app (e.g. Autodesk Maya) with incompatible versions.
```

> * USD Encoder:

Reviewer: We should add these new features:
* The USD encoder now writes the main shape attributes (e.g. start rule, random seed, OID) to the corresponding USD nodes.
* The USD encoder now writes the CGA report values to the corresponding USD component nodes.

Author: oops, good point! I'll add it

  * now uses the Callbacks interface to write the encoded files instead of directly wroiting to disk.
  * Added mesh merging option to control whether all meshes of one shape (i.e. building) should be merged by material or into one single mesh (option "meshMerging", values "perInitialShapeByMaterial" (default) and ,"perInitialShape")

>   * fixed a bug which led to a crash if windows network shares were used in a path via UNC (Universal Naming Convention).

Reviewer: ```suggestion
  * Fixed a bug which led to a crash if Windows network shares were used in a path via UNC (Universal Naming Convention).
```
